test/common/http/http2/codec_impl_test_util.h

```diff
@@ -26,7 +26,16 @@ class TestCodecSettingsProvider {
   void onSettingsFrame(const nghttp2_settings& settings_frame) {
     for (uint32_t i = 0; i < settings_frame.niv; ++i) {
       auto result = settings_.insert(settings_frame.iv[i]);
-      ASSERT(result.second);
+      // It is possible to have duplicate settings parameters, each new parameter replaces any
+      // existing value.
+      // https://tools.ietf.org/html/rfc7540#section-6.5
+      if (!result.second) {
+        ENVOY_LOG_MISC(debug, "Duplicated settings parameter {} with value {}",
+                       settings_frame.iv[i].settings_id, settings_frame.iv[i].value);
+        settings_.erase(result.first);
+        // Guaranteed success here.
+        ASSERT(settings_.insert(settings_frame.iv[i]).second);
```

Comment:
The body of `ASSERT`s can't be side effecting, since they disappear in opt build (and we do run tests on opt builds)..

Reply:
Oh, in hindsight that was pretty silly. Thank you!